modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -1878,7 +2007,7 @@ protected String encodeFID( PrimaryKey pkey, ResultSet rs, int offset ) throws S
         return encodeFID( keyValues );
     }
 
-    protected String encodeFID( List<Object> keyValues ) {
+    protected static String encodeFID( List<Object> keyValues ) {
```

Comment:
Normally classes that need services from JDBCDataStore take a reference to it (see how the various closeSafe methods are used)